test/model/dataTypes/FilePath.test.ts

```diff
@@ -0,0 +1,158 @@
+
+describe("FilePath", () => {
+  describe("getInstance()", () => {
+    it("should initialize the path value", () => {
+      expect(FilePath.getInstance("localFile.txt").value).toBe("localFile.txt");
+    });
+
+    it("should normalize .. and . segments of the path", () => {
+      const file1 = FilePath.getInstance("../../baseDir/one/.././");
+      expect(file1.value).toBe("../../baseDir/");
+    });
+
+    it("should normalize empty string to . (current working directory", () => {
+      const file1 = FilePath.getInstance("");
+      expect(file1.value).toBe(".");
+    });
+
+    it("should return the same instance of the FilePath for 2 separate invocations", () => {
+      const file1 = FilePath.getInstance("/someString.txt");
+      const file2 = FilePath.getInstance("/subdir/../someString.txt");
+      expect(file1).toBe(file2);
+    });
+
+    it("should detect a change in case sensitivity and throw an Error", () => {
+      expect(() => {
+        FilePath.getInstance("someString.txt");
+        FilePath.getInstance("SomeString.txt");
+      }).toThrow(
+        'FilePath "SomeString.txt" cannot overwrite equivalent file "someString.txt"'
+      );
+    });
+  });
+
+  describe("sortFilePaths()", () => {
+    it("should handle an empty array", () => {
+      expect(FilePath.sortFilePaths([])).toBeArrayOfSize(0);
+    });
+
+    it("should return the same instance", () => {
+      const arr = [FilePath.getInstance("./same")];
+      expect(FilePath.sortFilePaths(arr)).toBe(arr);
+    });
+
+    it("should handle duplicates", () => {
+      const arr = [
+        FilePath.getInstance("./same"),
+        FilePath.getInstance("./same"),
+      ];
+      expect(FilePath.sortFilePaths(arr)).toBe(arr);
+    });
+
+    it("should sort a mixed up array", () => {
+      const alpha = FilePath.getInstance("./alpha");
+      const bravo = FilePath.getInstance("./Bravo");
+      const charlie = FilePath.getInstance("./charlie");
+      const arr = [alpha, charlie, bravo];
+      expect(FilePath.sortFilePaths(arr)).toStrictEqual([
+        alpha,
+        bravo,
+        charlie,
+      ]);
+    });
+  });
+
+  describe("isAbsolute()", () => {
+    it("should return false for relative paths", () => {
+      const file = FilePath.getInstance("somedir/someString.txt");
+      expect(file.isAbsolute()).toBeFalse();
+    });
+
+    it("should return true for absolute paths", () => {
+      const file = FilePath.getInstance("/somedir/someString.txt");
+      expect(file.isAbsolute()).toBeTrue();
+    });
+  });
+
+  describe("isRelative()", () => {
+    it("should return true for relative paths", () => {
+      const file = FilePath.getInstance("somedir/someString.txt");
+      expect(file.isRelative()).toBeTrue();
+    });
+
+    it("should return true for absolute paths", () => {
+      const file = FilePath.getInstance("/somedir/someString.txt");
+      expect(file.isRelative()).toBeFalse();
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5966907)